### PR TITLE
Adding code for detective coat rack

### PR DESCRIPTION
**Pull request by DeityLink**
The sprite was already here, unused, I had to fix that.

coatrack

### CODE REVIEW

File: code/game/objects/structures/coatrack.dm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	fire_fuel = 3
+	var/obj/item/clothing/suit/storage/det_suit/suit = null
+	var/obj/item/clothing/head/det_hat/hat = null
+
+/obj/structure/coatrack/attack_hand(mob/user)
+	if(suit)
+		user << "<span class='notice'>You pick up the [suit] from the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		suit.loc = get_turf(src)
+		if(!usr.get_active_hand())
```
Comment (ComicIronic) [inline code/game/objects/structures/coatrack.dm]: Should be user.

File: code/game/objects/structures/coatrack.dm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		user.drop_item()
+		C.loc = src
+		suit = C
+		update_icon()
+
+	else if (istype(C, /obj/item/clothing/head/det_hat) && !hat)
+		user << "<span class='notice'>You place your [C] on the [src]</span>"
+		playsound(get_turf(src), "rustle", 50, 1, -5)
+		user.drop_item()
```
Comment (ComicIronic) [inline code/game/objects/structures/coatrack.dm]: For future reference, drop_item takes the argument of the loc you want to drop to. So you could write user.drop_item(src).

**Comment by d3athrow** [comment]
Something interesting you can do is use the pixel that he player clicked on to either grab the hat or the coat.

**Comment by DeityLink** [comment]
How can you do that?

**Comment by d3athrow** [comment]
looking into it it may just be for screen clicks but look at intent selection

**Pull request merged**